src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.cs

```diff
@@ -348,58 +349,48 @@ protected override void Dispose(bool disposing)
                 stateHandle = GCHandle.Alloc(state);
                 requestHandle = CreateRequestHandle(state, stateHandle);
                 state.RequestHandle = requestHandle;
-                needCleanup = true;
-
-                if (state.CancellationToken.IsCancellationRequested)
-                {
-                    state.TrySetCanceled(state.CancellationToken);
-                    return;
-                }
 
                 if (state.RequestMessage.Content != null)
                 {
                     Stream requestContentStream =
                         await state.RequestMessage.Content.ReadAsStreamAsync().ConfigureAwait(false);
+                    state.RequestContentStream = requestContentStream;
                     if (state.CancellationToken.IsCancellationRequested)
                     {
+                        RemoveEasyHandle(_multiHandle, stateHandle, state);
```

Comment:
We haven't added this handle yet, right?  Is this call here to instead clean up things like stateHandle?